### PR TITLE
RIP Nuage Cloudstack Plugin

### PR DESCRIPTION
**Pull request by fmaximus**
## Description

Removing the Nuage plugin as Nuage is discontinuing CloudStack support.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

- [x] Manual UI check
- [x] Smoke tests on Simulator
- [x] Upgrade from 4.11 after removing Nuage physical network

### CODE REVIEW

**Comment by GabrielBrascher** [comment]
@blueorangutan package

File: engine/schema/src/main/resources/META-INF/db/schema-41120to41200-cleanup.sql [hunk 1]
```diff
@@ -25,3 +25,7 @@ DROP TABLE IF EXISTS `cloud`.`iam_group_policy_map`;
 DROP TABLE IF EXISTS `cloud`.`iam_group`;
 DROP TABLE IF EXISTS `cloud`.`iam_policy_permission`;
 DROP TABLE IF EXISTS `cloud`.`iam_policy`;
+
+-- Cleanup Nuage VSP
+DELETE FROM `cloud`.`host` WHERE id in (SELECT vsp.host_id FROM `cloud`.`external_nuage_vsp_devices` vsp);
```
Comment (rhtyd) [inline engine/schema/src/main/resources/META-INF/db/schema-41120to41200-cleanup.sql]: @fmaximus please also remove `issueNuageVspResourceRequest` from role_permissions?

**Comment by rafaelweingartner** [comment]
Moved to 5.0.0. This PR is removing public APIs/features

**Comment by rhtyd** [comment]
@rafaelweingartner -1, deprecating or removing vendor/plugin specific `non-core` feature(s) and API(s) does not constitute a breaking of compatibility, warranting moving to 5.x. I'm +1 on removing the feature from master so if not 4.12.x, perhaps in 4.13.x, I'm sure we can put something in the docs around removal of features. Among the cloudstack plugins, the network and storage plugins are highly vendor dependent (and that said they may be developed/shipped outside of upstream project) and I think it should be up to Nuage how they want to support their users with cloudstack wrt a migration path.

**Comment by rafaelweingartner** [comment]
I know you hate the number 5 ;), but do not worry, we will have a lot of time to debate over this in the near future...

Right now, 4.12 is being released, and master is freezed.

**Comment by svenvogel** [comment]
@fmaximus i heared it from sales men from nuage/nokia that they want to focus on sd wan. its very bad decision i think but management think in a other way. good luck and thanks for help in the last years!

File: ui/scripts/configuration.js [hunk 2]
```diff
@@ -3001,9 +2996,9 @@
                                                                                 });
                                                                             }
                                                                         } else { //canenableindividualservice == true
-                                                                            if ($thisProviderDropdown.context.name in providerDropdownsForciblyChangedTogether) { //if this provider dropdown is one of provider dropdowns forcibly changed together earlier, make other forcibly changed provider dropdowns restore default option (i.e. 1st option in dropdown)
+                                                                            if (this.name in providerDropdownsForciblyChangedTogether) { //if this provider dropdown is one of provider dropdowns forcibly changed together earlier, make other forcibly changed provider dropdowns restore default option (i.e. 1st option in dropdown)
```
Comment (fmaximus) [inline ui/scripts/configuration.js]: I'll need to move this out to a separate bugfix.
Without this, Create Network Offering Popup doesn't work.
Related to upgrade of jquery.

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by fmaximus** [comment]
Rebased to deal with conflicts caused by merge of #3112

**Comment by rhtyd** [comment]
@fmaximus master is unfrozen now, sorry but can you fix the conflicts again. We can merge this after running a final regression test. Thanks.

**Comment by fmaximus** [comment]
I've solved the conflict.

**Comment by rhtyd** [comment]
Thanks @fmaximus - is this PR ready for testing/merging, the label WIP is not removed yet

@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test matrix

**Comment by krissterckx** [comment]
@rhtyd what is the plan for merging in this PR ? thanks

**Comment by rhtyd** [comment]
Thanks @fmaximus @krissterckx I'll merge this after the smoketests passes, let me kick them;
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test matrix

File: engine/schema/src/main/resources/META-INF/db/schema-41120to41200-cleanup.sql (hunk 1 above)
Comment (rhtyd) [reply engine/schema/src/main/resources/META-INF/db/schema-41120to41200-cleanup.sql]: cc @DaanHoogland @krissterckx  - the upgrade path needs to be in 4.12->4.13. I'll fix that while forward merging 4.11->4.12->master.

**Pull request merged**